src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -98,6 +100,15 @@ public void deletePatient(Patient target) {
         addressBook.removePatient(target);
     }
 
+    @Override
+    public Patient getPatient(IcNumber icNumber, List<Patient> patientList){
+        for (int i = 0; i < patientList.size(); i++) {
+            if(patientList.get(i).getIcNumber().equals(icNumber))
+                return patientList.get(i);
+        }
+        return null;
+    }
+
```

Comment:
I like how you implement your getPatient!! But if the patient could not be found, it could be quite dangerous to return a null. Do ensure that the other party calling this function checks for null. If it does receive a null, would be good to throw an exception!!! Could see my implementation of the exception PatientWithFieldNotFoundException. It has yet to be merged in code base but would be good if we could make use of this exception!!